bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/IntesisDynamicStateDescriptionProvider.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intesis.internal;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.binding.BaseDynamicStateDescriptionProvider;
+import org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionProvider;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+@Component(service = { DynamicStateDescriptionProvider.class, IntesisDynamicStateDescriptionProvider.class })
+@NonNullByDefault
+public class IntesisDynamicStateDescriptionProvider extends BaseDynamicStateDescriptionProvider {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final Map<ChannelUID, @Nullable StateDescription> descriptions = new ConcurrentHashMap<>();
+
+    public void setDescription(ChannelUID channelUID, @Nullable StateDescription description) {
+        logger.debug("Adding state description for channel {}", channelUID);
+        descriptions.put(channelUID, description);
+    }
+
+    public void removeAllDescriptions() {
+        logger.debug("Removing all state descriptions");
+        descriptions.clear();
+    }
+
+    @Override
+    public @Nullable StateDescription getStateDescription(Channel channel,
+            @Nullable StateDescription originalStateDescription, @Nullable Locale locale) {
+        StateDescription description = descriptions.get(channel.getUID());
+        return description;
+    }
+
+    @Override
+    @Deactivate
+    public void deactivate() {
+        descriptions.clear();
+    }
+}
```

Comment:
```suggestion
public class IntesisDynamicStateDescriptionProvider extends BaseDynamicStateDescriptionProvider {
}
```

And in `IntesisHomeHandler` replace
```
    updateStateDescription(thing.getChannel(channelId).getUID(), stateOptions);
```
with
```
     intesisStateDescriptionProvider.setStateOptions(thing.getChannel(channelId).getUID(), stateOptions);
```

And remove `updateStateDescription`.

Reply:
Thanks @Hilbrand, I will give it a try later tonight or tomorrow and report back.

Reply:
Thanks again @Hilbrand !!!! That worked an is implemented now. Had some trouble with the commit, but solved it.